internal/ast/ast.go

```diff
@@ -631,6 +631,15 @@ func (n *Node) Statements() []*Node {
 	return nil
 }
 
+func (n *Node) CanHaveStatements() bool {
```

Comment:
Alternatively, `Statements()` could return nil. I am unsure if that's a great idea.